src/torchpme/potentials/spline.py

```diff
@@ -74,6 +74,9 @@ def __init__(
         if len(y_grid) != len(r_grid):
             raise ValueError("Length of radial grid and value array mismatch.")
 
+        r_grid = r_grid.to(dtype=dtype, device=device)
+        y_grid = y_grid.to(dtype=dtype, device=device)
```

Comment:
Maybe out of the scope of this PR but question I have also to @ceriottm:

Shouldn't be the `device` and `dtype` deduced from the input and checked to be consistent and not given explicitly?

And why is the `device` and `dtype` not stored as private properties in the `Potential` classes. As we do it for the calculators? See 

https://github.com/lab-cosmo/torch-pme/blob/b3310fb132a41664dd80a33d0dac2d3dad20fe52/src/torchpme/calculators/calculator.py#L37-L40

Reply:
Yes, this is one of the things where there is a small mess throughout - some classes take these as arguments, some deduce them from the inputs, some store them and some don't. There is also quite a mess with "scalar arguments must be torch tensors vs just floats". I think we should have a quick (I dream) meeting to see what kind of API-changing cleanups should be done before release.

Reply:
Wouldn't touch this in this PR though which is a quickfix that is needed for #128

Reply:
Yes I agree 100%. I will open an issue related to these.